src/basic/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Basic Auth plugin.
+ */
+class BasicAuth {
+  constructor({ loginPath = '/login', logoutPath = '/logout', fetchUser = null } = {}) {
+    this.loginPath = loginPath;
+    this.logoutPath = logoutPath;
+    this.fetchUser = fetchUser;
+
+    if (!this.fetchUser) {
+      throw new Error('fetchUser is required');
+    }
+  }
+
+  init() {
+    return this.fetchUser()
+      .then(user => {
+        this.user = user;
+
+        return this.user;
+      })
+      .catch(() => false);
+  }
+
+  login() {
+    window.location = `${this.loginPath}?continue=${window.location.hash}`;
```

Comment:
if we use `window.location.hash` like this, there is a problem that I met here: https://github.com/OpenPaaS-Suite/esn/issues/16#issuecomment-672648286

Reply:
let's complete https://github.com/OpenPaaS-Suite/esn/issues/16#issuecomment-672648286 first then